File: BrickKit/bricks/src/main/java/com/wayfair/brickkit/models/TextViewModel.java
package com.wayfair.brickkit.models;

import com.wayfair.brickkit.brick.ViewModel;

import androidx.databinding.Bindable;

/**
 * A view model for a text brick.
 */
public class TextViewModel extends ViewModel<TextDataModel> {

    /**
     * Constructor.
     *
     * @param dataModel the {@link TextDataModel}
     */
    public TextViewModel(TextDataModel dataModel) {
        super(dataModel);
    }

    /**
     * Method that data binding uses to set the view's text.
     *
     * @return the text
     */
    @Bindable
    public String getText() {
        return getDataModel().getText();
    }
}

File: BrickKit/app/src/main/java/com/wayfair/brickkitdemo/simplebrick/viewmodel/TextViewModel.kt
package com.wayfair.brickkitdemo.simplebrick.viewmodel

import androidx.databinding.Bindable
import com.wayfair.brickkit.brick.ViewModel
import com.wayfair.brickkitdemo.simplebrick.datamodel.TextDataModel

/**
 * A view model for a text brick.
 */
class TextViewModel(dataModel: TextDataModel) : ViewModel<TextDataModel>(dataModel) {
    /**
     * Method that data binding uses to set the view's text.
     *
     * @return the text
     */
    @get:Bindable
    val text: String
        get() = dataModel.text
}
